node/impl/full/eth.go

```diff
@@ -1358,6 +1358,65 @@ type filterTipSetCollector interface {
 	TakeCollectedTipSets(context.Context) []types.TipSetKey
 }
 
+func ethLogFromEvent(entries []types.EventEntry) (data []byte, topics []ethtypes.EthHash, ok bool) {
+	var (
+		topicsFound      [4]bool
+		topicsFoundCount int
+		dataFound        bool
+	)
+	for _, entry := range entries {
+		// Check if the key is t1..t4
+		if len(entry.Key) == 2 && "t1" <= entry.Key && entry.Key <= "t4" {
```

Comment:
That would be kind of dangerous. E.g., if someone decides to extend this to `t12`, it'll break.

Reply:
You mean that the value being declared as a global variable will make it more prominent and subject to inadvertent changes? We can make it private to avoid programmatic mutations, but developer error can happen anywhere in the code. (not strongly held, we usually just define fixed arbitrary values as constants in most codebases)

Reply:
No, I'm concerned about code distance. I want anyone changing them to read the code that uses them.

Reply:
Ok.